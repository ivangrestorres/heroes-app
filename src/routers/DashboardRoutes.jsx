import React from "react";
import { Navbar } from "../components/ui/Navbar";
import { Switch, Route, Redirect } from "react-router-dom";
import { DcPage } from "../components/dc/DcPage";
import { MarvelPage } from "../components/marvel/MarvelPage";
import { HeroPage } from "../components/heroes/HeroPage";

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar></Navbar>
            <div className={"container mt-2"}>
                <Switch>
                    <Route exact path="/marvel" component={MarvelPage} />
                    <Route exact path="/heroe/:heroeId" component={HeroPage} />
                    <Route exact path="/dc" component={DcPage} />
                    <Redirect to="/marvel" />
                </Switch>
            </div>
        </>
    );
};
