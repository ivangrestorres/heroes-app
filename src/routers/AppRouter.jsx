import React from "react";
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
import { Navbar } from "../components/ui/Navbar";
import { LoginPage } from "../components/login/LoginPage";
import { DashboardRoutes } from "./DashboardRoutes";

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <Switch>
                    <Route exact path="/login" component={LoginPage} />
                    <Route path="/" component={DashboardRoutes} />
                </Switch>
            </div>
        </Router>
    );
};
