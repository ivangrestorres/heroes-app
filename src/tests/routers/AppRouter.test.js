import React from "react";
import "@testing-library/jest-dom";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { AppRouter } from "../../routers/AppRouter";
import { AuthContext } from "../../auth/AuthContext";
import { Navbar } from "../../components/ui/Navbar";

describe("Testing <AppRouter />", () => {
    const contextValue = {
        user: {
            logged: false,
        },
        dispatch: jest.fn(),
    };

    test("should show loggin if not authenticated", () => {
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test("should show marvel page if authenticated", () => {
        const contextValue = {
            user: {
                name: "Ivan",
                logged: true,
            },
            dispatch: jest.fn(),
        };

        const wrapper = mount(
            <MemoryRouter>
                <AuthContext.Provider value={contextValue}>
                    <AppRouter />
                </AuthContext.Provider>
            </MemoryRouter>
        );

        expect(wrapper.find(Navbar).exists()).toBe(true);
    });
});
