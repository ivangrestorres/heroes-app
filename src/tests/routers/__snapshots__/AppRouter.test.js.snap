// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <AppRouter /> should show loggin if not authenticated 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login Page
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
